txnkv/transaction/2pc.go

```diff
@@ -708,7 +741,10 @@ func (c *twoPhaseCommitter) initKeysAndMutations(ctx context.Context) error {
 
 func (c *twoPhaseCommitter) primary() []byte {
 	if len(c.primaryKey) == 0 {
-		return c.mutations.GetKey(0)
+		if c.mutations != nil {
+			return c.mutations.GetKey(0)
```

Comment:
As the `selectPrimaryForPessimisticLock` is introduced, maybe adding a check here to assert pessimistic transactions MUST have a valid primary key to make it safer?